business/noticeService.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get } = require('lodash')
+const DocBuilder = require('tiny-attribution-generator/lib/docbuilder').default
+const TextRenderer = require('tiny-attribution-generator/lib/outputs/text').default
+const TemplateRenderer = require('tiny-attribution-generator/lib/outputs/template').default
+const JsonSource = require('tiny-attribution-generator/lib/inputs/json').default
+
+const logger = require('../providers/logging/logger')
+
+class NoticeService {
+  constructor(definitionService, attachmentStore) {
+    this.definitionService = definitionService
+    this.attachmentStore = attachmentStore
+    this.logger = logger()
+  }
+
+  async generate(coordinates, template = '') {
+    const renderer = template ? new TemplateRenderer(template) : new TextRenderer()
+    const docbuilder = new DocBuilder(renderer)
+    const result = await this.definitionService.getAll(coordinates)
+    const packages = await Promise.all(
+      Object.keys(result).map(async id => {
+        const definition = result[id]
```

Comment:
what do we do when definitions are missing? the `getAll` above I think will return an entry but it will be largely empty. How/where do we accumulate all the errors etc? Would be good for the response to have that as well.

Reply:
let's implement all of chive and it's options

Reply:
let's return the json instead of the raw content with the summary of what happened

Reply:
problem buckets

 1. no license declared
 2. no tools run - meaning does not exist
 3. no copyright for component

Reply:
![image](https://user-images.githubusercontent.com/5168796/51414909-cfc5cf80-1b28-11e9-8046-711b9c91cc79.png)